tokio-trace/tokio-trace-core/src/lib.rs

```diff
@@ -0,0 +1,114 @@
+#![deny(missing_docs, warnings, missing_debug_implementations)]
+
+//! Core primitives for `tokio-trace`.
+//!
+//! `tokio-trace` is a framework for instrumenting Rust programs to collect
+//! structured, event-based diagnostic information. This crate defines the core
+//! primitives of `tokio-trace`.
+//!
+//! The crate provides:
+//!
+//! * [`Span`] identifies a span within the execution of a program.
+//!
+//! * [`Subscriber`], the trait implemented to collect trace data.
+//!
+//! * [`Metadata`] and [`Callsite`] provide information describing `Span`s.
+//!
+//! * [`Field`] and [`FieldSet`] describe and access the structured data attached to
+//!   a `Span`.
+//!
+//! * [`Dispatch`] allows span events to be dispatched to `Subscriber`s.
+//!
+//! Application authors will typically not use this crate directly. Instead,
+//! they will use the `tokio-trace` crate, which provides a much more
+//! fully-featured API. However, this crate's API will change very infrequently,
+//! so it may be used when dependencies must be very stable.
+//!
+//! [`Span`]: ::span::Span
+//! [`Subscriber`]: ::subscriber::Subscriber
+//! [`Metadata`]: ::metadata::Metadata
+//! [`Callsite`]: ::callsite::Callsite
+//! [`Field`]: ::field::Field
+//! [`FieldSet`]: ::field::FieldSet
+//! [`Dispatch`]: ::dispatcher::Dispatch
+//!
+
+#[macro_use]
+extern crate lazy_static;
+
+/// Statically constructs an [`Identifier`] for the provided [`Callsite`].
+///
+/// This may be used in contexts, such as static initializers, where the
+/// [`Callsite::id`] function is not currently usable.
+///
+/// [`Identifier`]: ::callsite::Identifier
+/// [`Callsite`]: ::callsite::Callsite
+/// [`Callsite::id`]: ::callsite::Callsite::id
+#[macro_export]
+macro_rules! identify_callsite {
```

Comment:
This is essentially just a stand-in for a `const fn` constructor for `callsite::Identifier`, so that it can be constructed publically even though the inner field is `#[doc(hidden)]` and it appears to be private in the docs. I'll add an example. 👍